db/db_range_del_test.cc

```diff
@@ -159,7 +159,7 @@ TEST_F(DBRangeDelTest, MaxCompactionBytesCutsOutputFiles) {
   // Want max_compaction_bytes to trigger the end of compaction output file, not
   // target_file_size_base, so make the latter much bigger
   opts.target_file_size_base = 100 * opts.max_compaction_bytes;
-  Reopen(opts);
+  DestroyAndReopen(opts);
```

Comment:
Just to double-check: is this related to this project (and if yes, how)?

Reply:
Reopening DB with a different comparator will cause writing to MANIFEST to fail: https://github.com/facebook/rocksdb/blob/master/db/version_set.cc#L4579. 

Without this PR, this `Reopen` succeeds, because the test never triggers a MANIFEST write.